docs/lazy-adr/adr-004-mvp-light-client.md

```diff
@@ -0,0 +1,292 @@
+diff --git a/types/light.go b/types/light.go
+--- a/types/light.go	(revision 64044aa2f2f2266d1476013595aa33bb274ba161)
++++ b/types/light.go	(date 1620481205049)
+@@ -13,6 +13,9 @@
+ type LightBlock struct {
+ 	*SignedHeader `json:"signed_header"`
+ 	ValidatorSet  *ValidatorSet `json:"validator_set"`
++
++	// DataAvailabilityHeader is only populated for DAS light clients for others it can be nil.
++	DataAvailabilityHeader *DataAvailabilityHeader `json:"data_availability_header"`
+ }
+```
+
+Alternatively, we could introduce a `DASLightBlock` that embeds a `LightBlock` and has the `DataAvailabilityHeader` as the only (non-optional) field.
+This would be more explict as it is a new type.
+Instead, adding a field to the existing `LightBlock`is backwards compatible and does not require any further code changes; the new type requires `To`- and `FromProto` functions at least.
+
+##### Provider
+
+The [`Provider`](https://github.com/tendermint/tendermint/blob/7f30bc96f014b27fbe74a546ea912740eabdda74/light/provider/provider.go#L9-L26) should be changed to additionally provide the `DataAvailabilityHeader` to enable DAS light clients.
+Implementations of the interface need to additionally to retrieve the `DataAvailabilityHeader` for the [modified LightBlock](#lightblock).
+Users of the provider need to indicate this to the provider.
+
+We could either augment the `LightBlock` method with a flag, add a new method solely for providing the `DataAvailabilityHeader`, or, we could introduce a new method for DAS light clients.
+
+The latter is preferable because it is the most explicit and clear, and it still keeps places where DAS is not used without any code changes.
+
+Hence:
+
+```diff
+Index: light/provider/provider.go
+===================================================================
+diff --git a/light/provider/provider.go b/light/provider/provider.go
+--- a/light/provider/provider.go	(revision 7d06ae28196e8765c9747aca9db7d2732f56cfc3)
++++ b/light/provider/provider.go	(date 1620298115962)
+@@ -21,6 +21,14 @@
+ 	// error is returned.
+ 	LightBlock(ctx context.Context, height int64) (*types.LightBlock, error)
+
++	// DASLightBlock returns the LightBlock containing the DataAvailabilityHeader.
++	// Other than including the DataAvailabilityHeader it behaves exactly the same
++	// as LightBlock.
++	//
++	// It can be used by DAS light clients.
++	DASLightBlock(ctx context.Context, height int64) (*types.LightBlock, error)
++
++
+ 	// ReportEvidence reports an evidence of misbehavior.
+ 	ReportEvidence(context.Context, types.Evidence) error
+ }
+```
+Alternatively, with the exact same result, we could embed the existing `Provider` into a new interface: e.g. `DASProvider`that add this method.
+This is completely equivalent as above and which approach is better will become more clear when we spent more time on the implementation.
+
+Regular light clients will call `LightBlock` and DAS light clients will call `DASLightBlock`.
```

Comment:
Why DAS light client is not a regular light client if LL is based on DAS in the first place?